src/index.js

```diff
@@ -162,7 +166,16 @@ module.exports = (config = {}) => {
       return validate(options.json, conf.schema.req)
       .then(() => validate(options.json, conf.schema.query))
       .then(() => extendOptions(conf, options))
-      .then((extendedOptions) => client.request(extendedOptions));
+      .then((extendedOptions) => {
+        if (name === 'githubLogin') {
```

Comment:
I would love to use a more generic approach here so we do not need to add new conditionals for every new auth method. WDYT?
```suggestion
        const token = resp.auth.client_token
        if (token) {
          client.token = token
          return resp
        }
```